src/snapshot.zig

```diff
@@ -0,0 +1,159 @@
+const std = @import("std");
+const base = @import("base.zig");
+const parse = @import("check/parse.zig");
+
+var verbose_log: bool = false;
+
+fn log(comptime fmt: []const u8, args: anytype) void {
+    if (verbose_log) {
+        std.log.info(fmt, args);
+    }
+}
+
+pub fn main() !void {
+    var gpa_impl = std.heap.GeneralPurposeAllocator(.{}){};
+    defer {
+        _ = gpa_impl.deinit();
+    }
+    const gpa = gpa_impl.allocator();
+
+    const args = try std.process.argsAlloc(gpa);
+    defer std.process.argsFree(gpa, args);
+
+    // Check for verbose flag
+    var target_snapshot: ?[]const u8 = null;
+
+    for (args[1..]) |arg| {
+        if (std.mem.eql(u8, arg, "--verbose")) {
+            verbose_log = true;
+        } else if (target_snapshot == null) {
+            target_snapshot = arg;
+        }
```

Comment:
This feels like wrong behaviour. It throws out all early args and just takes the very last arg. If you pass in 5 args, I would assume we should snapshot test all 5.

Reply:
Good idea, I didn't think of that

Reply:
Added in 537dc3db49